File: clients/Zwoo.BotDashboard.Distributor/src/Configuration.cs
namespace Zwoo.BotDashboard.Distributor;

/// <summary>
/// Represents the configuration for the distributor.
/// </summary>
public class Configuration
{
    /// <summary>
    /// Creates a new configuration from environment variables.
    /// </summary>
    /// <returns>the configuration</returns>
    public static Configuration FromEnv()
    {
        return new Configuration
        {
            Id = Environment.GetEnvironmentVariable("ZRP_DEBUG_ID") ?? string.Empty,
            Instance = Environment.GetEnvironmentVariable("ZRP_DEBUG_INSTANCE") ?? string.Empty,
            ProxyUrl = Environment.GetEnvironmentVariable("ZRP_DEBUG_PROXY_URL") ?? string.Empty
        };
    }

    /// <summary>
    /// The distributor ID.
    /// </summary>
    public required string Id { get; set; }

    /// <summary>
    /// The distributor instance (the url of the distributor).
    /// </summary>
    public required string Instance { get; set; }

    /// <summary>
    /// The url of the zrp proxy to connect to.
    /// </summary>
    public required string ProxyUrl { get; set; }

    private Configuration() { }
}

File: clients/dashboard-client/src/configuration.ts
export class Configuration {
  /**
   * create a new Configuration instance from the import.meta.env
   *
   * @static
   * @returns {Configuration}
   *
   * @memberOf Configuration
   */
  public static fromImportMeta(): Configuration {
    return new Configuration(import.meta.env.ZRP_DEBUG_PROXY_URL!);
  }

  /**
   * create a new Configuration instance from the process.env
   *
   * @static
   * @returns {Configuration}
   *
   * @memberOf Configuration
   */
  public static fromEnv(): Configuration {
    return new Configuration(process.env.ZRP_DEBUG_PROXY_URL!);
  }

  /**
   * create a new Configuration instance from the given proxyUrl
   *
   * @static
   * @param {string} proxyUrl the zrp proxy url
   * @returns {Configuration}
   *
   * @memberOf Configuration
   */
  public static withUrl(proxyUrl: string): Configuration {
    return new Configuration(proxyUrl);
  }

  private constructor(public readonly proxyUrl: string) {}
}
